debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbConnector.java

```diff
@@ -223,6 +225,32 @@ public Config validate(Map<String, String> connectorConfigs) {
             try (ConnectionContext connContext = new ConnectionContext(config)) {
                 try (MongoClient client = connContext.clientFor(connContext.hosts())) {
                     client.listDatabaseNames();
+                    // Oplog mode is not supported for MongoDB 5+
+                    // The version string format is not guaranteed so defensive measures are in place
+                    final Document versionDocument = client.getDatabase("local")
+                            .runCommand(new Document("buildInfo", 1));
+                    if (versionDocument != null) {
+                        final String versionString = versionDocument.getString("version");
+                        if (versionString != null) {
+                            final String[] versionComponents = versionString.split("\\.");
+                            if (versionComponents.length > 0) {
+                                try {
+                                    final int majorVersion = Integer.parseInt(versionComponents[0]);
+                                    final ConfigValue captureModeValue = results
+                                            .get(MongoDbConnectorConfig.CAPTURE_MODE.name());
+                                    final String captureMode = config.getString(MongoDbConnectorConfig.CAPTURE_MODE);
```

Comment:
Wouldn't it be better to construct a `MongoDbConnectorConfig()` and call the typed accessor for this prop?